pakyow-core/lib/pakyow/core/router.rb

```diff
@@ -196,6 +196,15 @@ class Router
 
     # @api private
     def initialize(context)
+      if router = context.previous_router_instance
+        # copy state from previous router to the new one
+        # this is so that state is shared when rerouting from one router to another one
```

Comment:
Nice. I wish we could avoid this, but I think its best/most convenient for users to be able to think that instance variables in the router will stick around for the entire request/response cycle. It would be too much to ask them to use a method to store state like: store(:foo, :bar), and I guess we don't want to evaluate the route block in the controller context.

Anyway, just thinking out-loud. 

:+1: